db/view/view.cc

```diff
@@ -2360,24 +2360,28 @@ update_backlog node_update_backlog::add_fetch(unsigned shard, update_backlog bac
     return std::max(backlog, _max.load(std::memory_order_relaxed));
 }
 
-future<bool> check_view_build_ongoing(db::system_distributed_keyspace& sys_dist_ks, const sstring& ks_name, const sstring& cf_name) {
-    return sys_dist_ks.view_status(ks_name, cf_name).then([] (std::unordered_map<locator::host_id, sstring>&& view_statuses) {
-        return boost::algorithm::any_of(view_statuses | boost::adaptors::map_values, [] (const sstring& view_status) {
-            return view_status == "STARTED";
+future<bool> check_view_build_ongoing(db::system_distributed_keyspace& sys_dist_ks, const locator::token_metadata& tm, const sstring& ks_name,
+        const sstring& cf_name) {
+    using view_statuses_type = std::unordered_map<locator::host_id, sstring>;
+    return sys_dist_ks.view_status(ks_name, cf_name).then([&tm] (view_statuses_type&& view_statuses) {
+        return boost::algorithm::any_of(view_statuses, [&tm] (const view_statuses_type::value_type& view_status) {
+            // Only consider status of known hosts.
+            return tm.get_endpoint_for_host_id(view_status.first) && view_status.second == "STARTED";
         });
```

Comment:
we could have inverted the order, so tm.get_endpoint_for_host_id(view_status.first) is only called for entries with STARTED state. get_endpoint_for_host_id() performs a linear search on all endpoints.

Reply:
not that I think it will bring any significant improvement in practice though. just caught my eyes :-)

Reply:
I will send a v3 with the order swapped.

Reply:
Done in v3.